cloudwatch_logs_common/src/utils/cloudwatch_facade.cpp

```diff
@@ -197,19 +233,32 @@ Aws::CloudWatchLogs::ROSCloudWatchLogsErrors CloudWatchFacade::GetLogStreamToken
     for (auto it = log_stream_list.begin(); it != log_stream_list.end(); ++it) {
       Aws::CloudWatchLogs::Model::LogStream curr_log_stream = *it;
       if (curr_log_stream.GetLogStreamName().c_str() == log_stream) {
```

Comment:
This is only checking the first page of results. The [API](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_DescribeLogStreams.html) doesn't ensure order results, so we should check all pages

Reply:
modified to check all pages.